test/test_utils.py

```diff
@@ -414,6 +424,52 @@ def test_is_in_or_equal_fuzzer(path_1, path_2):
         pytest.fail(f"Exception raised: {e}")
 
 
+@settings(derandomize=os.getenv("CI") is not None)
+@given(
+    path=create_path_string(),
+    blocked_paths=create_path_list(),
+    allowed_paths=create_path_list(),
+    file_sets=st.lists(create_path_set(), min_size=0, max_size=3),
+)
+def test_is_allowed_file_fuzzer(
```

Comment:
Cool idea to use the fuzzer here. But if I'm understanding correctly, randomly generating paths is very, very unlikely to produce any overlaps between the parameters, right? So this is actually unlikely to test what happens if a `path` is in `blocked_paths` or `allowed_paths` or `file_sets`? Perhaps part of the path should be manually specified or the values should be chosen from a set with smaller cardinality

Reply:
I greatly reduced the cardinality of the allowed possible paths set and also added some corner cases manually in a different test.